addons/binding/org.openhab.binding.verisure/README.md

```diff
@@ -0,0 +1,359 @@
+| siteName       | String                | This channel reports the name of the site.                 |
+| siteId         | Number                | This channel reports the site ID of the site.              |
+ 
+### Verisure Siren
+
+#### Configuration Options
+
+*   deviceId - Device Id
+    *   Sensor Id. Example 5A4C_35FT (Note: the '_' in the Verisure ID, found in the Verisure App or My Pages)
+
+#### Channels
+
+([siren]) supports the following channels:
+ 
+| Channel Type ID | Item Type            | Description                                                | 
+|----------------|-----------------------|------------------------------------------------------------|                                                                                                                                          
+| humidity       | Number                | This channel reports the current humidity in percentage.   |
+| temperature    | Number:Temperature    | This channel reports the current humidity in percentage.   |
+| lastUpdate     | String                | This channel reports the last time this sensor was updated.|
+| location       | String                | This channel reports the location.                         |
+| siteName       | String                | This channel reports the name of the site.                 |
+| siteId         | Number                | This channel reports the site ID of the site.              |
+
+### Verisure Night Control
+
+#### Configuration Options
+
+*   deviceId - Device Id
+    *   Sensor Id. Example 5A4C_35FT (Note: the '_' in the Verisure ID, found in the Verisure App or My Pages)
+
+#### Channels
+
+([nightControl]) supports the following channels:
+ 
+| Channel Type ID | Item Type            | Description                                                | 
+|----------------|-----------------------|------------------------------------------------------------|                                                                                                                                          
+| humidity       | Number                | This channel reports the current humidity in percentage.   |
+| temperature    | Number:Temperature    | This channel reports the current humidity in percentage.   |
+| lastUpdate     | String                | This channel reports the last time this sensor was updated.|
+| location       | String                | This channel reports the location.                         |
+| siteName       | String                | This channel reports the name of the site.                 |
+| siteId         | Number                | This channel reports the site ID of the site.              |
+
+### Verisure Yaleman SmartLock
+
+#### Configuration Options
+
+*   deviceId - Device Id
+    *   Sensor Id. Example 5A4C_35FT (Note: the '_' in the Verisure ID, found in the Verisure App or My Pages)
```

Comment:
Is there reason why underscore is required? JSON doesn't contain it, so why it should be present id deviceId? It was required earlier because of thing name cant contain spaces, but deviceId could. In fact binding could remove white space/underscores/etc in both deviceId and JSON before make comparison.

Reply:
I think that I inherited that, JSON contains spaces for all Verisure IDs except for SmartLock for some reason. The way is is printed on the boxes is with space, but when you install a new device you can enter the 8 characters either with or without a space.

What should we use if we decide to change it?

Reply:
@paulianttila  I've fixed and pushed the README comments, but I've not done anything to the '_' question above.

Reply:
I pushed up a commit that makes the binding handle non system administrator accounts.

@paulianttila Should we change the representation of the Verisure Id or should we stick with having them separated by a "_"?

Reply:
I pushed up a community reported issue of handling special characters in password.

Reply:
I pushed up a commit with updated README to reflect actual support for channel humidity for some climate devices.

Reply:
I pushed up a community reported issue about support for channel humidity for climate devices. The binding now supports the channel humidity for the curremt implementation of humudity in Verisure devices. I also changed siteId to update a long value.

Reply:
I pushed up a fix to support user presence for many users.

Reply:
About underscore, i would implemented so that user could give as many spaces or undescores to device id and when binding parse json data I would remove all special character from both device id from thing configuration and from json data. This should be safe as device id is a serial number of the device in hexadecimal format (except verisure alarm).

conf.deviceId.replaceAll("[^a-zA-Z]+", "").equalsIgnoreCase(json.deviceId.replaceAll("[^a-zA-Z]+", ""))

WDYT?

Reply:
Thing type ids alarm, broadbandConnection and userPresence lacks a Verisure ID. 
For alarm and broadbandConnection I've used the pattern "thing type id"_installationNumber as deviceId, e.g. `alarm_1` and `broadbandConnection_2`. 

Should I change that to `alarm1` and `broadbandConnection2`?
The user could then for manually configuring  thing use `alarm_1` since the new logic would remove it.

For userPresence I yesterday got community feedback that the binding did not support tracking more than one user.
I fixed that by implementing a new name scheme to create a unique deviceId by using the user's eMail address but exchanging the `@` with `_`, e.g. `userpresence_jannegpriv_gmail_com_1`
meaning myself at installation 1.

Should I change that to `userpresencejannegprivgmailcom1`?  Or maybe remove userpresence?
`jannegprivgmailcom1`
Or should we keep the `@`sign? 

WDYT?

Then for devices with a Verisure Id using autoconfig, I would remove all spaces and generate a deviceId looking like `ABCD1234`.
If user was configuring the thing manually, the user could use all of the following definitions for deviceId; `ABCD1234`,`ABCD 1234` and `ABCD_1234`.

Correct?

Reply:
FYI: I got an eMail from Verisure that they will upgrade their webpage and app in the coming weeks, this will probably affect the implementation of the binding.

Reply:
@paulianttila I got these errors tonight when trying to build locally with maven:
````
[ERROR] Code Analysis Tool has found: 
 3 error(s)! 
 12 warning(s) 
 0 info(s)
[ERROR] .binding.verisure/ESH-INF/binding/binding.xml:[5]
Cannot find the declaration of element binding:binding.
[ERROR] .binding.verisure/ESH-INF/thing/thing-types.xml:[5]
Cannot find the declaration of element thing:thing-descriptions.
[ERROR] .binding.verisure.internal.VerisureHandlerFactory.xml:[3]
There were whitespace characters used for indentation. Please use tab characters instead
````

Is this due to the changes with the maven builds and the new binding package?

Reply:
Not directly. Its due to the schema name space changes in the XML files: #5026

Reply:
@Hilbrand 
How can I fix this error since the file is auto-generated?

````
[ERROR] .binding.verisure.internal.VerisureHandlerFactory.xml:[3]
There were whitespace characters used for indentation. Please use tab characters instead
````